22_pipeslab/pipeslab.txt

```diff
@@ -3,19 +3,42 @@ For each of the following, type a single command line that performs:
   1. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner.
   Answer: 
 
+  tocode@tocode-VirtualBox:~$ ls -alF | grep tocode | cut -c4 | grep x | wc | cut -d' ' -f6
+  20
+  grep x is not must
+  
   2. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner, group and others.
   Answer: 
-
+ 
+  ls -alF | grep tocode | grep x..x..x | wc | cut -d' ' -f6
+  
+  
   3. Print the number of lines from "ls -l" which represent paths that were modified today (no matter which day we run the command)
   Answer: 
 
+  ls -alF | grep tocode | grep Jan..2 | wc | cut -d' ' -f7
+  
   4. Modify (1) and (2) to count files only (no directories)
   Answer:
+  
+	ls -alF | grep tocode | grep -v / | cut -c4 | grep x | wc | cut -d' ' -f7
+	
+	ls -alF | grep tocode | grep -v / | grep x..x..x | wc | cut -d' ' -f7
 
+	
   5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. Remember not to show the special loopback address 127.0.0.1
+	
+	date:
+	ls -alF | grep "$(date '+%b %e')"
+	
+	ifconfig | grep broadcast | tr -s ' ' | cut -d' ' -f3
```

Comment:
that's what I get on my machine:
```
vitali@Tux:~/tmp$ ifconfig 
lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:420957 errors:0 dropped:0 overruns:0 frame:0
          TX packets:420957 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1 
          RX bytes:142306734 (142.3 MB)  TX bytes:142306734 (142.3 MB)

wlp1s0    Link encap:Ethernet  HWaddr 48:45:20:2d:e9:bb  
          inet addr:10.0.0.11  Bcast:10.0.0.255  Mask:255.255.255.0
          inet6 addr: fe80::8a94:1e71:55d1:45cb/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:413878922 errors:0 dropped:0 overruns:0 frame:0
          TX packets:177838135 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:584336617718 (584.3 GB)  TX bytes:17583095780 (17.5 GB)

vitali@Tux:~/tmp$ cat /proc/version
Linux version 4.4.0-53-generic (buildd@lcy01-28) (gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.4) ) #74-Ubuntu SMP Fri Dec 2 15:59:10 UTC 2016
```

```ifconfig``` is a horrible command when it comes to parsing it's output.. it's very inconsistent between unix systems, and even between distros...